src/Microsoft.Health.Dicom.Blob/Extensions/AzureBlobExportOptionsExtensions.cs

```diff
@@ -16,17 +18,26 @@ internal static class AzureBlobExportOptionsExtensions
 {
     public static async Task<BlobContainerClient> GetBlobContainerClientAsync(
         this AzureBlobExportOptions exportOptions,
-        IExportIdentityProvider identityProvider,
+        IServerCredentialProvider credentialProvider,
         BlobClientOptions options,
         CancellationToken cancellationToken = default)
     {
         EnsureArg.IsNotNull(exportOptions, nameof(exportOptions));
         EnsureArg.IsNotNull(options, nameof(options));
 
         if (exportOptions.BlobContainerUri != null)
-            return exportOptions.UseManagedIdentity
-                ? new BlobContainerClient(exportOptions.BlobContainerUri, await identityProvider.GetCredentialAsync(cancellationToken))
-                : new BlobContainerClient(exportOptions.BlobContainerUri);
+        {
+            if (exportOptions.UseManagedIdentity)
+            {
+                TokenCredential credential = await credentialProvider.GetCredentialAsync(cancellationToken);
+                if (credential == null)
+                    throw new InvalidOperationException(DicomBlobResource.MissingServerIdentity);
```

Comment:
If a user doesn't configure an identity, we'll throw here. As an aside, I should probably add `InvalidOperationException` to our middleware's list of 400s